code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -447,3 +447,47 @@
 	//Possibly burning a bit of organic matter through maillard reaction, so a *tiny* bit more heat would be understandable
 	air.set_temperature(air.return_temperature() + cleaned_air * 0.002)
 	SSresearch.science_tech.add_point_type(TECHWEB_POINT_TYPE_DEFAULT, cleaned_air*MIASMA_RESEARCH_AMOUNT)//Turns out the burning of miasma is kinda interesting to scientists
+
+
+/datum/gas_reaction/stim_ball
+	priority = 7
+	name ="Stimulum Energy Ball"
+	id = "stimball"
+
+/datum/gas_reaction/stim_ball/init_reqs()
+	min_requirements = list(
+		/datum/gas/pluoxium = STIM_BALL_MOLES_REQUIRED,
+		/datum/gas/stimulum = STIM_BALL_MOLES_REQUIRED,
+		/datum/gas/plasma = STIM_BALL_MOLES_REQUIRED,
+		"TEMP" = FIRE_MINIMUM_TEMPERATURE_TO_EXIST
+	)
+
+/// Reaction that burns stimulum and plouxium into radballs and partial constituent gases, but also catalyzes the combustion of plasma.
+/datum/gas_reaction/stim_ball/react(datum/gas_mixture/air, datum/holder)
+	var/turf/open/location
+	var/old_heat_capacity = air.heat_capacity()
+	if(istype(holder,/datum/pipeline)) //Find the tile the reaction is occuring on, or a random part of the network if it's a pipenet.
+		var/datum/pipeline/pipenet = holder
+		location = get_turf(pick(pipenet.members))
+	else
+		location = get_turf(holder)
+	var/reaction_rate = min(STIM_BALL_MAX_REACT_RATE, air.get_moles(/datum/gas/pluoxium), air.get_moles(/datum/gas/stimulum), air.get_moles(/datum/gas/plasma))
+	var/balls_shot = round(reaction_rate/STIM_BALL_MOLES_REQUIRED)
+	//A percentage of plasma is burned during the reaction that is converted into energy and radballs, though mostly pure heat. 
+	var/plasma_burned = QUANTIZE((air.get_moles(/datum/gas/plasma) + 5*reaction_rate)*STIM_BALL_PLASMA_COEFFICIENT)
+	//Stimulum has a lot of stored energy, and breaking it up releases some of it. Plasma is also partially converted into energy in the process.
+	var/energy_released = (reaction_rate*STIMULUM_HEAT_SCALE) + (plasma_burned*STIM_BALL_PLASMA_ENERGY)
+	air.adjust_moles(/datum/gas/stimulum, -reaction_rate)
+	air.adjust_moles(/datum/gas/pluoxium, -reaction_rate)
+	air.adjust_moles(/datum/gas/nitryl, reaction_rate*5)
+	air.adjust_moles(/datum/gas/plasma, -plasma_burned)
+	if(balls_shot>0)
+		var/angular_increment = 360/balls_shot
+		var/random_starting_angle = rand(0,360)
+		for(var/i in 1 to balls_shot)
+			location.fire_nuclear_particle((i*angular_increment+random_starting_angle)%360)
```

Comment:
Whats the goal of the modulo 360 here?

Reply:
wanted to sanitize the input angle before passing it in, but after testing with values above 360 it looks like it doesn't need to be modulo'd since it does it by itself.